heartbeat/docker-compose

```diff
@@ -0,0 +1,297 @@
+: ${OCF_RESKEY_ymlfile=${OCF_RESKEY_ymlfile_default}}
+
+USAGE="Usage: $0 {start|stop|status|monitor|validate-all|meta-data}"
+
+##########################################################################
+
+usage()
+{
+	echo $USAGE >&2
+}
+
+meta_data()
+{
+cat <<END
+<?xml version="1.0"?>
+<!DOCTYPE resource-agent SYSTEM "ra-api-1.dtd">
+<resource-agent name="docker-compose">
+<version>1.0.1</version>
+<longdesc lang="en">
+Manages docker services using docker-compose as an OCF resource in an High Availability setup.
+It relies on a well-tested docker compose YAML file which distributed on an identical location on all cluster nodes.
+
+Caveat: 1. A YAML file (docker-compose.yml) and an optional Dockerfile
+           must be provided in a working directory.
+        2. It is suggested to test run the docker-compose and verify on all cluster nodes
+           before enabling this agent.
+</longdesc>
+<shortdesc lang="en">This script manages docker services using docker-compose.</shortdesc> 
+
+<parameters>
+
+<parameter name="binpath">
+<longdesc lang="en">
+The docker-composer binary path.
+For example, "/usr/bin/docker-compose"
+</longdesc>
+<shortdesc lang="en">The docker-composer binary path</shortdesc>
+<content type="string" default="$OCF_RESKEY_binpath_default"/>
+</parameter>
+
+<parameter name="dirpath" required="1">
+<longdesc lang="en">
+The directory contains docker compose yaml file.
+For example, "/data/docker"
+</longdesc>
+<shortdesc lang="en">Directory contains docker compose files</shortdesc>
+<content type="string"/>
+</parameter>
+
+<parameter name="ymlfile">
+<longdesc lang="en">
+The docker-compose yaml file.
+For example, "docker-compose.yml"
+</longdesc>
+<shortdesc lang="en">The docker compose yaml</shortdesc>
+<content type="string" default="$OCF_RESKEY_ymlfile_default"/>
+</parameter>
+
+<parameter name="svcname" required="1">
+<longdesc lang="en">
+The docker service name.
+For example, "myservice"
+</longdesc>
+<shortdesc lang="en">The docker service name</shortdesc>
+<content type="string"/>
+</parameter>
+
+</parameters>
+
+<actions>
+<action name="start" timeout="240s"/>
+<action name="stop" timeout="20s"/>
+<action name="monitor" depth="0" timeout="10s" interval="60s" />
+<action name="validate-all" timeout="5s" />
+<action name="meta-data"  timeout="5s"/>
+</actions>
+</resource-agent>
+END
+exit $OCF_SUCCESS
+}
+
+if [ -r "$OCF_RESKEY_binpath" -x "$OCF_RESKEY_binpath" ]; then
+	COMMAND="$OCF_RESKEY_binpath"
+else
+	COMMAND=$(which docker-compose 2>/dev/null)
+fi
+
+DIR="$OCF_RESKEY_dirpath"
+YML="$OCF_RESKEY_ymlfile"
+SVC="$OCF_RESKEY_svcname"
+
+docker_kill()
+{
+	for i in $(docker ps --all | awk -e '$NF ~ /\<'"$SVC"'_.*_[0-9]+\>/ {print $1}'); do
+		docker kill $i &>/dev/null
+		docker rm $i &>/dev/null || RTV=false
+	done
+	if [ "$RTV" = false ]; then
+		ocf_log err "failed to kill docker"
+		return $OCF_ERR_GENERIC
+	else
+		RUN=false
+	fi
+}
+
+docker-compose_status()
```

Comment:
It seems like CI fails due to the -'s in the function names (only _'s are supported in function names from what I recall).